valhalla/baldr/graphtile.h

```diff
@@ -588,7 +588,20 @@ class GraphTile : public boost::intrusive_ref_counter<GraphTile, GraphTileRefCou
     if (!invalid_time && (flow_mask & kPredictedFlowMask) && de->has_predicted_speed()) {
       seconds %= midgard::kSecondsPerWeek;
       uint32_t idx = de - directededges_;
-      float speed = predictedspeeds_.speed(idx, seconds);
+      uint32_t bucket = seconds / kSpeedBucketSizeSeconds;
+#ifdef ENABLE_THREAD_SAFE_TILE_REF_COUNT
+      float speed = predictedspeeds_.speed(idx, bucket);
+#else
+      float speed;
+      {
+        BucketSpeed& entry = predicted_speed_cache[idx];
```

Comment:
and yeah as you point out its very unfortunate that we remove the "read-only" nature of graphtile here. if we want to do something like this maybe we need to keep the cache in the graph reader or somehwere else so that we can preserve graphtiles `const` interface without resorting to `mutable`